### ISSUE TITLE
Error in .local

### ISSUE THREAD

**Issue by aldojongejan** [LTLA/SingleR#56]
Dear all,
Installed the developers version (1.1.0), but get the following error when trying to get the Immune Cell Expression values: 
devtools::install_github('LTLA/SingleR')
library(SingleR)
immCellExpr <- DatabaseImmuneCellExpressionData()
**snapshotDate(): 2019-04-29
Error in .local(x, i, j, ...) : 'i' must be length 1**

Any ideas?

I have attached the sessionInfo:
[sessionInfo_SingleR.txt](https://github.com/LTLA/SingleR/files/3818431/sessionInfo_SingleR.txt)

Thanks in advance!
kind regards,
Aldo

**Comment by LTLA** [comment]
Follow the official protocol for installing the devel version:

https://bioconductor.org/packages/devel/bioc/html/SingleR.html

**Comment by aldojongejan** [comment]
Yeah, you’re fast! ;-)

This would mean I have to install R4.0, no possibility to run it under R3.6.1?

Regards,
Aldo

**Comment by LTLA** [comment]
The immediate answer to your technical problem is that it is possible to run it under 3.6, but you have to update all your other packages. For example, for me:

```
> library(SingleR)
> immCellExpr <- DatabaseImmuneCellExpressionData()
snapshotDate(): 2019-10-22
see ?SingleR and browseVignettes('SingleR') for documentation
loading from cache
see ?SingleR and browseVignettes('SingleR') for documentation
loading from cache
```

Notice the more recent snapshot date compared to yours.

However, if you want to run under 3.6, you should just use the BioC-release version. Unless you know better, but if you're the type of person who knows better, installing R 4.0.0 should not be a problem.

**Comment by aldojongejan** [comment]
Dear Aaron,

Thanks for your fast answers!

I do want to install R4.0, but according to its homepage, 3.6.1 is still the latest versions (at least that is what I get to see…see screenshot below). Or do I miss something?
I definitely not the one that knows better ;-)

I indeed saw you more recent date of the snapshot, which puzzled me as I just today downloaded SingleR and ran the command…
So, probably something else is not up to date with my system and I am having trouble with old caches..

Thanks for your help!

Aldo

[cid:image002.jpg@01D5954B.131E0D00]

**Comment by LTLA** [comment]
R 4.0.0 is R-devel, see https://stat.ethz.ch/R/daily/.

You need to make sure *all* of your BioC packages are up to date, not just _SingleR_. Make sure `BiocManager::version()` reports "3.10" and `BiocManager::valid()` doesn't report any problems.

**Comment by aldojongejan** [comment]
OK, I sure need more coffee …. I am very ashamed for wasting your time! ;-)

Regards,
Aldo

**Comment by LTLA** [comment]
No worries. Going to sleep now.

**Comment by aldojongejan** [comment]
;-)

Really appreciate your help and all the good packages!

Aldo

**Comment by LTLA** [comment]
I'll assume that this was fixed, but if not, the discussion at davismcc/scater#100 may be pertinent.

**Comment by aldojongejan** [comment]
Yes, it was fixed!
I got everything working. Thanks again, Aaron!

Regards,
Aldo

**Comment by tufeiyinjie** [comment]
I encountered an error when I run the below code on a remote cluster.
>library(SingleR)
>ref <- celldex::HumanPrimaryCellAtlasData()

error information:
 "snapshotDate(): 2020-04-27
Error in .local(x, i, j = j, ...) : 'i' must be length 1". 

Here is my sessioninfo.
R version 4.0.0 (2020-04-24)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Red Hat Enterprise Linux

Matrix products: default
BLAS:   /usr/lib64/libblas.so.3.4.2
LAPACK: /usr/lib64/liblapack.so.3.4.2

However, when I run the same code on the local Rstudio, it can generate the object "ref" and the running information is also listed below.
> ref <- celldex::HumanPrimaryCellAtlasData()
snapshotDate(): 2022-10-31
see ?celldex and browseVignettes('celldex') for documentation
loading from cache
see ?celldex and browseVignettes('celldex') for documentation
loading from cache

I want to fix the error that appeared on the cluster. I also run the "BiocManager::valid()", which didn't generate any problems but show up warning information.
warning message:
670 packages out-of-date; 38 packages too new.
I run BiocManager::install() to update the package but no package was updated.

Any information is appricated.

Fei
